packages/metro-inspector-proxy/src/InspectorProxy.js

```diff
@@ -162,16 +162,30 @@ class InspectorProxy {
     // $FlowFixMe[value-as-type]
     wss.on('connection', async (socket: WS, req) => {
       try {
+        const fallbackDeviceId = String(this._deviceCounter++);
+
         const query = url.parse(req.url || '', true).query || {};
+        const deviceId = query.device || fallbackDeviceId;
         const deviceName = query.name || 'Unknown';
         const appName = query.app || 'Unknown';
-        const deviceId = this._deviceCounter++;
+
+        const oldDevice = this._devices.get(deviceId);
+        if (oldDevice) {
+          // Keep the debugger connection alive when disconnecting the device, if possible
```

Comment:
Does the `new Device` created below somehow inherit `_debuggerConnection` from `oldDevice`?

Also, `_debuggerConnection` is a private variable (by naming convention) and ideally shouldn't be accessed outside of the `Device` class

Reply:
> Does the `new Device` created below somehow inherit `_debuggerConnection` from `oldDevice`?

It does, but we might need to add some logic when swapping out the `Device` instances. Basically, in my tests I found that the open debugger automatically triggered the `Device.handleDebuggerConnection` method (background reconnection). This sets the debugger connection to the new `Device` instance.

I think an ideal scenario is to just invoke that method directly during the initialization.

Originally, I had to unset the debugger connection in order to avoid the [device's socket close event handler](https://github.com/facebook/metro/blob/main/packages/metro-inspector-proxy/src/Device.js#L123-L127).

Reply:
> Also, `_debuggerConnection` is a private variable (by naming convention) and ideally shouldn't be accessed outside of the `Device` class

One thing we can do is move the shutdown logic into a handler on the device. With that, we don't need to access `_debuggerConnection`, `_name`, and `_app`. Does that make sense to you? @voideanvalue

Reply:
Yup makes sense, I think, feel free to modify the `Device` class (as it's private within this package) to accomplish the new logic needed as well-named methods.